platform/broadcom/docker-syncd-brcm/syncd.sh

```diff
@@ -1,12 +1,122 @@
 #!/usr/bin/env bash
+#
+# Based on /etc/init.d/syncd
 
-function clean_up {
-    service syncd stop
-    exit
+CMD_SYNCD=/usr/bin/syncd
+
+CMD_DSSERVE=/usr/bin/dsserve
+CMD_DSSERVE_ARGS="$CMD_SYNCD --diag"
+
+ENABLE_SAITHRIFT=0
+
+PLATFORM_DIR=/usr/share/sonic/platform
+HWSKU_DIR=/usr/share/sonic/hwsku
+
+if [ -x $CMD_DSSERVE ]; then
+    CMD=$CMD_DSSERVE
+    CMD_ARGS=$CMD_DSSERVE_ARGS
+else
+    CMD=$CMD_SYNCD
+fi
+
+parse_yaml() {
+    local prefix=$2
+    local s='[[:space:]]*' w='[a-zA-Z0-9_]*' fs=$(echo @|tr @ '\034')
+    sed -ne "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
+         -e "s|^\($s\)\($w\)$s:$s\(.*\)$s\$|\1$fs\2$fs\3|p"  $1 |
+    awk -F$fs '{
+        indent = length($1)/2;
+        vname[indent] = $2;
+        for (i in vname) {if (i > indent) {delete vname[i]}}
+        if (length($3) > 0) {
+           vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
+           printf("%s%s%s=\"%s\"\n", "'$prefix'",vn, $2, $3);
+        }
+     }'
+}
+
+start_bcm()
+{
+    [ -e /dev/linux-bcm-knet ] || mknod /dev/linux-bcm-knet c 122 0
+    [ -e /dev/linux-user-bde ] || mknod /dev/linux-user-bde c 126 0
+    [ -e /dev/linux-kernel-bde ] || mknod /dev/linux-kernel-bde c 127 0
+}
+
+start_mlnx()
```

Comment:
this is for mellanox platform. can we consider to still put this syncd.sh into the sairedis repo?